x-pack/plugins/rule_registry/server/utils/create_persistence_rule_type_wrapper.ts

```diff
@@ -146,6 +156,174 @@ export const createPersistenceRuleTypeWrapper: CreatePersistenceRuleTypeWrapper
                 return { createdAlerts: [], errors: {}, alertsWereTruncated: false };
               }
             },
+            alertWithSuppression: async (
```

Comment:
**question:** Could this function be combined within `alertWithPersistence` by adding a `suppression` optional property? It would facilitate moving this code to the framework soon by having a single code path. The `alertWithPersistence ` function could take the following: 
```
{
  alerts: Array<{ _id: string; _source: T; }>;
  refresh: boolean | 'wait_for';
  maxAlerts?: number;
  enrichAlerts?: ...;
  ...
  suppression?: {
    window: string;
    currentTimeOverride?: Date;
  };
}
```

Reply:
Right now the difficulty with combining them into a single function would be that the `T` in `_source: T` for `alertWithSuppression` extends `SuppressionFieldsLatest` whereas for `alertWithPersistence` it has no restrictions. `alertWithSuppression` relies on the `extends` so it can rely on the input docs already being grouped and annotated with the suppression metadata like docs count, group end time, and instance ID.

We could pull some of the enrichment and shared augmentation logic out into helper functions though if that would ease the future transition? The deduplication and update logic in `alertWithSuppression` are unique to suppression so a large portion of this function would still be needed in branching logic if we combine them into one "service" function.

Reply:
Thanks for the explanation! Yeah, let's keep it simple and maybe pull some shared code into helper functions, that will definitely help future transition.